src/mumudvb.c

```diff
@@ -169,30 +184,38 @@ int dont_send_scrambled=0;
 extern log_params_t log_params;
 
 // prototypes
-static void SignalHandler (int signum);//below
+#ifndef DISABLE_DVB_API
+static void SignalHandler (int signum); //below
+#endif
 int read_multicast_configuration(multi_p_t *, mumudvb_channel_t *, char *); //in multicast.c
 void init_multicast_v(multi_p_t *multi_p); //in multicast.c
 
 void chan_new_pmt(unsigned char *ts_packet, mumu_chan_p_t *chan_p, int pid);
 
 int processt2(unsigned char* input_buf, unsigned int input_buf_offset, unsigned char* output_buf, unsigned int output_buf_offset, unsigned int output_buf_size, uint8_t plpId);
 
-int
-main (int argc, char **argv)
+int main (int argc, char **argv)
 {
 	// file descriptors
-	fds_t fds; /** File descriptors associated with the card */
+	static fds_t fds; /** File descriptors associated with the card */
 	memset(&fds,0,sizeof(fds_t));
 
+#ifdef _WIN32
+	WSADATA wsaData;
+
+	if (WSAStartup(MAKEWORD(2, 2), &wsaData) != 0)
+		return -1;
+#endif
+
 	//Thread information
-	pthread_t signalpowerthread=0;
+	pthread_t signalpowerthread = pthread_self();
```

Comment:
official (and proper) way to compare threads is using the pthread api, instead of checking pthread_t (which is opaque) to 0/NULL